datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedTaskCountEstimator.java

```diff
@@ -27,6 +41,22 @@
    */
   public int getTaskCount(ClusterThroughputInfo throughputInfo, List<String> assignedPartitions,
       List<String> unassignedPartitions) {
-    throw new NotImplementedException();
+    Validate.notNull(throughputInfo, "null throughputInfo");
+    Validate.notNull(assignedPartitions, "null assignedPartitions");
+    Validate.notNull(unassignedPartitions, "null unassignedPartitions");
+
+    Map<String, PartitionThroughputInfo> throughputMap = throughputInfo.getPartitionInfoMap();
+
+    Set<String> allPartitions = new HashSet<>(assignedPartitions);
+    allPartitions.addAll(unassignedPartitions);
+
+    // total throughput in KB/sec
+    int totalThroughput = throughputMap.entrySet().stream().
+        filter(e -> allPartitions.contains(e.getKey())).mapToInt(e -> e.getValue().getBytesInKBRate()).sum();
+    LOG.debug("Total throughput in all partitions: {}KB/sec", totalThroughput);
+
+    int taskCountEstimate = (int) Math.ceil((double) totalThroughput / (TASK_CAPACITY_MBPS_DEFAULT * 1024));
```

Comment:
do we know if the KBps rate we get from the provider is KB or KiB as there are a couple of conventions going around with that and based on it, we might have to change 1024 to 1000?

Reply:
The field name suggests that it is rate in KB/sec, hence 1024.